Configure CodeMirror editor for Web Handlers configuration

Amnish04 opened the pull request:
In this PR, I have:

1. Replaced the [TextArea](https://chakra-ui.com/docs/components/textarea/usage) that was being used to configure **Web Handlers** with [CodeMirror](https://github.com/uiwjs/react-codemirror) code editor to ensure a good developer experience.

![image](https://github.com/tarasglek/chatcraft.org/assets/78865303/9c0234b9-ce33-48c0-8352-6997ed284833)

2. Also added some onboarding instructions so users can easily understand how this feature works.
This instructions can be hidden/shown with a switch I've added right above the editor.

![image](https://github.com/tarasglek/chatcraft.org/assets/78865303/445da8a3-bf96-4f50-9e40-faf298f5dece)


3. We, of course, also get all the functionalities `CodeHeader` component implements - including **copying**, **downloading** and opening the code in a new browser tab.

This closes #540

Amnish04 commented:
I just added a new pref to our Settings called `showWebHandlersInstructions`.

I am storing it in local storage so familiar users don't have to turn it off on each refresh.

tarasglek commented:
I don't think you need to hide instructions and ui for that. Just have instructions in default value

humphd commented on src/components/WebHandlersConfigModal.tsx:
```diff
@@ -79,30 +116,84 @@ function WebHandlersConfigModal({ isOpen, onClose, finalFocusRef }: WebHandlersC
 
   return (
-    <Modal isOpen={isOpen} onClose={onModalClose} size="lg" finalFocusRef={finalFocusRef}>
+    <Modal isOpen={isOpen} onClose={onModalClose} size="2xl" finalFocusRef={finalFocusRef}>
```
Doest he modal have a prop that allows resizing?

Amnish04 replied:
The modal is resizable by default. I think this property is to specify the max width. Since the addition of comments was demanding more space to be user friendly, I increased it.

humphd commented on src/components/WebHandlersConfigModal.tsx:
```diff
@@ -79,30 +116,84 @@ function WebHandlersConfigModal({ isOpen, onClose, finalFocusRef }: WebHandlersC
-            </FormControl>
+            <Text textAlign={isMobile ? "justify" : "start"}>
+              {/* eslint-disable-next-line react/no-unescaped-entities */}
+              If you want to extend ChatCraft's functionality by plugging external services your own
```
"If you want to extend ChatCraft's default URL handling by adding your own external services..."

Amnish04 replied:
Just fixed the message

Amnish04 commented:
> I don't think you need to hide instructions and ui for that. Just have instructions in default value

@tarasglek I added that since I found it a little unnecessary to have to scroll below all that info even when you don't need it.

But maybe its an overkill. Just removed it.

humphd commented:
Can we reduce the width of the instructions comment, so it doesn't cause horizontal scrolling?


https://github.com/tarasglek/chatcraft.org/assets/427398/14a86748-81c9-4ab5-a058-ea6a1d44b2d5

Amnish04 commented:
@humphd Weird, there is not horizontal scrolling on my side 🤔
![image](https://github.com/tarasglek/chatcraft.org/assets/78865303/ab487556-9d98-4b33-9f5d-a04e42dac178)

Amnish04 commented:
@humphd What's the width of your modal in pixels?

Amnish04 commented:
> 672px

Huh, its same for me. I wonder what could be causing it?

humphd commented:
If you remove some of the extra `###` characters, which aren't needed, it will be fine.

Amnish04 commented:
> If you remove some of the extra `###` characters, which aren't needed, it will be fine.

Just pushed. Should work well now
<img width="685" alt="image" src="https://github.com/tarasglek/chatcraft.org/assets/78865303/940c25f9-360e-4b0f-83dc-d7ca1193b8c5">

Amnish04 commented:
@humphd I was also thinking we should probably store the default web handler in memory instead of local storage.

```yaml
- handlerUrl: https://taras-scrape2md.web.val.run/
  method: GET
  matchPattern: ^https:\/\/\S+$
```

And add it at the end of the list so it takes the least precedence.

Right now, if the user deletes it, it's gone forever until they clear their local storage and the hook add it back for them

tarasglek commented:
can also put the docs under config :)

Amnish04 commented:
> can also put the docs under config :)

@tarasglek I just added the sample handler at the end as a comment. I am not sure if that's how you wanted but we can always have a follow up issue to make any changes.

humphd commented:
@Amnish04 I would go with this and improve in follow-ups. What you have is awesome imho.
